OpenRA.Game/Server/Server.cs

```diff
@@ -308,9 +308,9 @@ void ValidateClient(Connection newConn, string data)
 					Slot = LobbyInfo.FirstEmptySlot(),
 					PreferredColor = handshake.Client.Color,
 					Color = handshake.Client.Color,
-					Faction = "Random",
+					Faction = handshake.Client.Faction,
```

Comment:
You'll need to validate these first to prevent malicious players from sending garbage to kill the server.